src/tree/values/nodes/Heading.js

```diff
@@ -0,0 +1,55 @@
+import Node from "./Node";
+/**
+ * A header in a document.
+ */
+class Heading extends Node {
+	/**
+	 * Makes a new header object.
+	 *
+	 * @param {number} startIndex			The index in the original text-string where this Node begins.
+	 * @param {number} endIndex			The index in the original text-string where this Node ends.
+	 * @param {number} level				The header level (e.g. 1 for main heading, 2 for subheading lvl 2, etc.)
+	 * @param {TextContainer} textContainer	The text contained within this heading.
+	 */
+	constructor( startIndex, endIndex, level, textContainer ) {
+		super( "heading", startIndex, endIndex );
+		this.level = level;
+		this.textContainer = textContainer;
+	}
+
+	/**
+	 * Makes a clone of this Node.
+	 *
+	 * @returns {Node} The clone of this Node.
+	 */
+	clone() {
+		return new Heading(
+			this.startIndex,
+			this.endIndex,
+			this.level,
+			this.textContainer,
+		);
+	}
+
+	/**
+	 * Maps the given function to this Heading.
+	 *
+	 * @param {mapCallback} mappingFunction	The function that should be mapped to this Heading.
+	 * @returns {Node} A clone of this Heading after the given function has been applied to it.
+	 */
+	map( mappingFunction ) {
```

Comment:
I am having troubles with understanding how these `map` and `filter` functions are going to be used in the real parser (and sadly the specs do not allow to understand that better). Could you elaborate on that in the JSDocs or maybe add a more realistic spec? Maybe it is better to write these functions (if needed) when we know better what the parser needs.